src/engine/ExportQueryExecutionTrees.cpp

```diff
@@ -467,11 +467,19 @@ ExportQueryExecutionTrees::idToStringAndTypeForEncodedValue(Id id) {
         }
         double dIntPart;
         // If the fractional part is zero, write number with one decimal place
-        // to make it distinct from integers. Otherwise, use `%g`, which uses
+        // to make it distinct from integers. Otherwise, use `%.13g`, which uses
         // fixed-size or exponential notation, whichever is more compact.
-        std::string out = std::modf(d, &dIntPart) == 0.0
-                              ? absl::StrFormat("%.1f", d)
-                              : absl::StrFormat("%g", d);
+        std::string out;
+        if (std::modf(d, &dIntPart) == 0.0) {
+          out = absl::StrFormat("%.1f", d);
+        } else {
+          out = absl::StrFormat("%.13g", d);
+          // For some values `modf` evaluates to zero, but rounding still leads
+          // to a value without a trailing '.0'.
```

Comment:
I don't understand this comment. We are in the `else` part here, so `std::modf` did not evaluate to zero. Why then does the comment refer to the case where it evaluates to zero?

Reply:
`std::modf` returns 0 when the double/float _is_ cleanly divisible by 1. But the formatter might choose to not render that fraction, even if `std::modf` returns a non zero value. These cases are meant by that.